core/trino-main/src/main/java/io/trino/execution/CreateViewTask.java

```diff
@@ -76,6 +80,19 @@ public String getName()
 
         accessControl.checkCanCreateView(session.toSecurityContext(), name);
 
+        if (metadata.isView(session, name) && !statement.isReplace()) {
+            throw semanticException(ALREADY_EXISTS, statement, "View already exists: '%s'", name);
+        }
+        if (metadata.isMaterializedView(session, name)) {
+            throw semanticException(ALREADY_EXISTS, statement, "Materialized view already exists: '%s'", name);
+        }
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, name);
+        // Normally, when a table with the same name does not exist but a view with the same name exists
+        // tableHandle should return empty. But some connectors are exceptions, eg. hive connector...
+        if (tableHandle.isPresent() && !metadata.isView(session, name)) {
+            throw semanticException(TABLE_ALREADY_EXISTS, statement, "Table already exists: '%s'", name);
+        }
+
```

Comment:
this generally looks good to me.
One small comment is that we could cache
`metadata.isView(session, name)` result in the local variable and then skip the `isMaterializedView` and `getTableHandle` calls if it's true (i.e. if it's a view it is definitely not a materialized view nor a table). This would both avoid unnecessary overhead and allow you to drop the view check in the `tableHandle.isPresent()` if statement.

Reply:
I will do it